### PR TITLE
[fix][broker] fix incorrect unack stats when dup ack a message

### LINKED ISSUE #19268: [Bug] Acknowledge the same message-id causes inconsistency unackedMessages number.
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

all

### Minimal reproduce step

```java
/**
     * This test is used to test the unackedMessage number in the consumer.
     */
    @Test
    public void testDuplicateAcknowledgement() throws PulsarClientException, PulsarAdminException {
        final String topicName = "persistent://public/default/duplicated-acknowledgement-test";
        @Cleanup
        Producer<byte[]> producer = pulsarClient.newProducer()
                .topic(topicName)
                .create();
        @Cleanup
        Consumer<byte[]> consumer1 = pulsarClient.newConsumer()
                .topic(topicName)
                .subscriptionName("sub-1")
                .acknowledgmentGroupTime(0, TimeUnit.SECONDS)
                .subscriptionType(SubscriptionType.Shared)
                .subscribe();
        producer.send("1".getBytes(StandardCharsets.UTF_8));
        Message<byte[]> message = consumer1.receive();
        consumer1.acknowledge(message);
        consumer1.acknowledge(message);
        assertEquals(admin.topics().getStats(topicName).getSubscriptions()
                .get("sub-1").getUnackedMessages(), 0);
    }
```

### What did you expect to see?

`unackedMessages=0`.

### What did you see instead?

`unackedMessages=-1`.

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by labuladong**
Fixes #19268

### Modifications

For shared-type consumers, if can't find the message position in pendingAck set, skip the following process.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [x] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/labuladong/pulsar/pull/20

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java [hunk 1]
```diff
@@ -473,6 +473,11 @@ private CompletableFuture<Long> individualAckNormal(CommandAck ack, Map<String,
             long ackedCount = 0;
             long batchSize = getBatchSize(msgId);
             Consumer ackOwnerConsumer = getAckOwnerConsumer(msgId.getLedgerId(), msgId.getEntryId());
+            if (Subscription.isIndividualAckMode(ackOwnerConsumer.subType())
+                && !ackOwnerConsumer.getPendingAcks().containsKey(msgId.getLedgerId(), msgId.getEntryId())) {
+                // this message has been acked, skip to keep stats correct
+                continue;
+            }
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java:480]: The `ackOwnerConsumer` can be a different consumer. 

```java
private Consumer getAckOwnerConsumer(long ledgerId, long entryId) {
    Consumer ackOwnerConsumer = this;
    if (Subscription.isIndividualAckMode(subType)) {
        if (!getPendingAcks().containsKey(ledgerId, entryId)) {
            for (Consumer consumer : subscription.getConsumers()) {
                if (consumer != this && consumer.getPendingAcks().containsKey(ledgerId, entryId)) {
                    ackOwnerConsumer = consumer;
                    break;
                }
            }
        }
    }
    return ackOwnerConsumer;
}
```

We will still have the issue that the unacked messages count is decreased for this consumer, but actually the owner consumer is another consumer.

**Review by codelipenghui — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java (hunk 1 above)
Comment (labuladong) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java:480]: I don't understand, if the actual consumer is another consumer, `addAndGetUnAckedMsgs` can update the unack count of that consumer:

https://github.com/apache/pulsar/blob/96fb7dab7a1bc260d8ee3c9d5df15de6863ee15b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java#L1049-L1060

**Comment by poorbarcode** [comment]
Since we will start the RC version of `3.0.0` on `2023-04-11`, I will change the label/milestone of PR who have not been merged.
- The PR of type `feature` is deferred to `3.1.0`
- The PR of type `fix` is deferred to `3.0.1`

So drag this PR to `3.0.1`

**Comment by nodece** [comment]
https://github.com/apache/pulsar/issues/19268 has been fixed by https://github.com/apache/pulsar/pull/20990, so close this PR.

**Pull request closed without merging**